File: app/src/main/java/com/sakhand/downloadmanager/social/SocialDetector.kt
package com.sakhand.downloadmanager.social

import com.sakhand.downloadmanager.data.SocialPlatform

/**
 * تشخیص خودکار پلتفرم از روی لینک
 */
object SocialDetector {

    fun detect(url: String): SocialPlatform? {
        val u = url.trim().lowercase()
        return when {
            "youtube.com" in u || "youtu.be" in u || "youtube-nocookie.com" in u -> SocialPlatform.YOUTUBE
            "instagram.com" in u || "instagr.am" in u -> SocialPlatform.INSTAGRAM
            "pinterest.com" in u || "pin.it" in u -> SocialPlatform.PINTEREST
            else -> null
        }
    }
}

File: windows/SakhandDm/Core/SocialDetector.cs
namespace SakhandDm.Core;

/// <summary>تشخیص خودکار پلتفرم از روی لینک</summary>
public static class SocialDetector
{
    public static SocialPlatform? Detect(string? url)
    {
        if (string.IsNullOrWhiteSpace(url)) return null;
        var u = url.Trim().ToLowerInvariant();
        if (u.Contains("youtube.com") || u.Contains("youtu.be") || u.Contains("youtube-nocookie.com"))
            return SocialPlatform.Youtube;
        if (u.Contains("instagram.com") || u.Contains("instagr.am"))
            return SocialPlatform.Instagram;
        if (u.Contains("pinterest.com") || u.Contains("pin.it"))
            return SocialPlatform.Pinterest;
        return null;
    }
}
